TheraLang.BLL/Services/UserService.cs

```diff
@@ -0,0 +1,128 @@
+﻿using AutoMapper;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using TheraLang.BLL.DataTransferObjects;
+using TheraLang.BLL.Interfaces;
+using TheraLang.DAL.Entities;
+using TheraLang.DAL.UnitOfWork;
+
+namespace TheraLang.BLL.Services
+{
+    public class UserService : IUserService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IFileService _fileService;
+
+        public UserService(IUnitOfWork unitOfWork, IFileService fileService)
+        {
+            _unitOfWork = unitOfWork;
+            _fileService = fileService;
+        }
+
+        public async Task<UserAllDto> GetMyProfile(Guid id)
+        {
+            try
+            {
+                var user = await _unitOfWork.Repository<User>().Get(u => u.Id == id);
+                var userDetails = await _unitOfWork.Repository<UserDetails>().Get(u => u.UserDetailsId == id);
+
+                var detailsMapper = new MapperConfiguration(cfg => cfg.CreateMap<UserDetails, UserAllDto>()).CreateMapper();
+                var userAll = detailsMapper.Map<UserDetails, UserAllDto>(userDetails);
+
+                var mapper = new MapperConfiguration(cfg => cfg.CreateMap<User, UserAllDto>()).CreateMapper();
+                var userInf = mapper.Map<User, UserAllDto>(user);
+                userAll.Email = userInf.Email;
+
+                return userAll;
+            }
+            catch (Exception ex)
+            {
+                throw new Exception($"Error when getting user by {nameof(id)}={id}: ", ex);
+            }
+        }
+
+        public async Task<UserDetailsDto> GetUserById(Guid id)
```

Comment:
better specify in name that it return  UserDetails

Reply:
Ok

Reply:
done